dev-docs/RFCs/v7.x/python-integration-rfc.md

```diff
@@ -0,0 +1,117 @@
+
+* **Author**: Ib Green, Andrew Duberstein
+* **Date**: Mar, 2019
+* **Status**: **Draft**
+
+## Summary
+
+This RFC proposes adding a minimal official Python integration to deck.gl, focusing on Jupyter notebook support.
+
+
+## Background
+
+Improved support for data science and machine learning visualization is a focus area for deck.gl, and deck.gl is already very useful for geospatial visualizations in these contexts. Many data scientists work in Python and being able to effortlessly visualize data generated by libraries like numpy and pandas directly from the notebook is very valuable.
+
+By providing a set of very simple bindings, focusing on overcoming the most basic integration issues and exposing a reasonable subset of the deck.gl API, we should be able to provide significantly increase adoption and value of deck.gl in these areas.
+
+Offering such an integration clearly makes sense, as long as we can do it with low effort and maintenance cost.
+
+
+### Leveraging the deck.gl JSON API
+
+Normally, exposing an API in a new language requires porting that API to the new language, which can be a significant undertaking, creating a large set of class and function "wrappers" that glue to the original code, with associated maintenance costs.
+
+However, deck.gl now supports a JSON API. The JSON API makes it possible for users to specify deck.gl layers and access a large subset of deck.gl's functionality without directly using the deck.gl JavaScript API, making it an ideal API entry point to expose in other languages.
+
+The idea is thus to create a minimal binding in Python that allows a python application to generate a JSON payload compatible with the deck.gl API and then start an inline frame running deck.gl rendering that JSON payload.
+
+
+### Focus Areas
+
+To minimize initial effort and long-term maintenance, we want to keep the scope very limited and focus on solving core problems such as:
+
+* Making the deck.gl Python module easy to install and use for Python users
+* Handle proper initial sizing and resizing the JavaScript/HTML window
+* Efficiently transfer large data sets between Python and JavaScript
+* Make it easy to build more ambitions Python class libraries on top of the basic module.
+
+
+### Constraints
+
+deck.gl should not grow to include a major Python port of its API (e.g. a Python class library that mirrors all deck.gl layers). That would represent too big of a distraction to the core deck.gl mission. We should leave such work to the community, and instead encourage, support and promote such efforts as they happen.
+
+Something we would likely want to support is that other deck.gl/Python integtation libraries build on our official integration (i.e. they use it as a python dependency). Alternatively, they could just use it as an inspiration/springboard to get started, or fork it to help them solve some of the initial integration issues.
+
+
+## Technical Challenges
+
+* Starting up a JavaScript iframe in Notebook
```

Comment:
Worth mentioning that the solution will be loading pre-bundled deck.gl from CDN?

We do not publish the JSON module as part of the standalone bundle in 6.4. In 7.0 we plan to publish a master bundle (deckgl.min.js) as well as individual bundles for each submodule. We need to make the decision which bundles to include for the Python API.

Reply:
Thanks. Added these comments into the text, together with a comment that we might be able to make the selection of layer modules bundles configurable by Python code.